src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -52,16 +77,61 @@ function useChatUtilsHook(): UseChatUtilsHook {
     });
   };
 
-  const goToChatScreen = (channel: ChannelList) => {
+  const goToCommunityScreen = (channel: ChannelList) => {
     setChannelAsRead(channel);
-    if (channel?.channelType === 'ANON_POST_NOTIFICATION')
-      return helperGoToPostDetailScreen(channel);
 
-    navigation.navigate('SampleChatScreen');
+    const topicName = channel?.name?.replace(/^#/, '');
+    const navigationParam = {
+      id: convertTopicNameToTopicPageScreenParam(topicName)
+    };
+
+    navigation.navigate('TopicPageScreen', navigationParam);
+  };
+
+  const openChat = (screen: string) => {
+    navigation.dispatch({
+      ...CommonActions.reset({
+        routes: [
+          {
+            name: 'AuthenticatedStack',
+            params: {
+              screen: 'HomeTabs',
+              params: {
+                screen: 'ChannelList',
+                isReset: true
+              }
+            }
+          },
+          {
+            name: 'AuthenticatedStack',
+            params: {
+              screen,
+              isReset: true
+            }
+          }
+        ]
+      })
+    });
+  };
+
+  const goToChatScreen = (channel: ChannelList, from) => {
     setChat({
       ...chat,
       selectedChannel: channel
     });
+    setChannelAsRead(channel);
+    if (channel?.channelType === ANON_PM) {
+      if (from === GROUP_INFO) {
+        return openChat('SampleChatScreen');
+      }
+      navigation.navigate('SampleChatScreen');
+    } else {
+      if (from === GROUP_INFO) {
+        return openChat('SignedChatScreen');
+      }
+      navigation.navigate('SignedChatScreen');
+    }
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this section, you're using a string literal 'GROUP_INFO' to compare with the `from` parameter. It's better to use constants for such comparisons to avoid potential bugs due to typos and improve maintainability. If 'GROUP_INFO' is used elsewhere in your code, consider defining it as a constant.



<!-- This is an auto-generated comment by OSS CodeRabbit -->